pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+        for prop in properties['common-properties']:
+            properties['datapoints'][idx][prop] = properties['common-properties'][prop]
+
+    # compression time doesn't belong in common-properties
+    properties['common-properties'].pop('compression-time', None)
+
+    filename_yaml = os.path.splitext(os.path.basename(filename_xml))[0] + '.yaml'
+
+    # add path
+    filename_yaml = os.path.join(output, filename_yaml)
+
+    with open(filename_yaml, 'w') as outfile:
+        outfile.write(yaml.dump(properties, default_flow_style=False))
+    print('Converted to ' + filename_yaml)
+
+    # now validate
+    ChemKED(yaml_file=filename_yaml)
+
+    return filename_yaml
+
+
+def convert_to_ReSpecTh(filename_ck, output_path=''):
+    """Convert ChemKED file to ReSpecTh XML file.
+
+    This converter uses common information in a ChemKED file to generate a
+    ReSpecTh XML file. Note that some information may be lost, as ChemKED stores
+    some additional attributes.
+
+    Arguments:
+        filename_ck (`str`): Filename of existing ChemKED YAML file to be converted.
+        output_path (`str`, optional): Path for output ReSpecTh XML file.
+    """
+    c = ChemKED(yaml_file=filename_ck)
+
+    root = etree.Element('experiment')
+
+    file_author = etree.SubElement(root, 'fileAuthor')
+    file_author.text = c.file_author['name']
+
+    # right now ChemKED just uses an integer file version
+    file_version = etree.SubElement(root, 'fileVersion')
+    major_version = etree.SubElement(file_version, 'major')
+    major_version.text = str(c.file_version)
+    minor_version = etree.SubElement(file_version, 'minor')
+    minor_version.text = '0'
+
+    respecth_version = etree.SubElement(root, 'ReSpecThVersion')
+    major_version = etree.SubElement(respecth_version, 'major')
+    major_version.text = '1'
+    minor_version = etree.SubElement(respecth_version, 'minor')
+    minor_version.text = '0'
+
+    # Only ignition delay currently supported
+    exp = etree.SubElement(root, 'experimentType')
+    if c.experiment_type == 'ignition delay':
+        exp.text = 'Ignition delay measurement'
+    else:
+        raise NotImplementedError('Only ignition delay type supported for conversion.')
+
+    reference = etree.SubElement(root, 'bibliographyLink')
+    citation = ''
+    for author in c.reference.authors:
+        citation += author['name'] + ', '
+    citation += (c.reference.journal + ' (' + str(c.reference.year) + ') ' +
+                 str(c.reference.volume) + ':' + c.reference.pages + '. ' + c.reference.detail
+                 )
+    reference.set('preferredKey', citation)
+    reference.set('doi', c.reference.doi)
+
+    apparatus = etree.SubElement(root, 'apparatus')
+    kind = etree.SubElement(apparatus, 'kind')
+    kind.text = c.apparatus.kind
+
+    common_properties = etree.SubElement(root, 'commonProperties')
+    # ChemKED objects have no common properties once loaded... can we check for properties
```

Comment:
I don't think we need to check... the point of common properties is to ensure that values are consistent, but they're guaranteed to be consistent by using the script here so it shouldn't be a problem.